File: Longest Binary Subsequence Less Than or Equal to K/Longest Binary Subsequence Less Than or Equal to K.java
// Solution for Longest Binary Subsequence Less Than or Equal to K in JAVA

class Solution {
    public int longestSubsequence(String s, int k) {
        int n = s.length();
        long powerOfTwo = 1;  
        int maxLength = 0;       

        for (int i = n - 1; i >= 0; i--) {
            if (s.charAt(i) == '0') {
                maxLength++;
            } else if (powerOfTwo <= k) {
                k -= powerOfTwo;
                maxLength++;
            }
            if (powerOfTwo <= k) {
                powerOfTwo <<= 1; 
            }
        }
        return maxLength;
    }
}

File: Longest Binary Subsequence Less Than or Equal to K/Longest Binary Subsequence Less Than or Equal to K.py
# Solution for Longest Binary Subsequence Less Than or Equal to K in PY

class Solution:
    def longestSubsequence(self, s: str, k: int) -> int:
        n = len(s)
        powerOfTwo = 1
        maxLength = 0

        for i in range(n - 1, -1, -1):
            if s[i] == '0':
                maxLength += 1
            elif powerOfTwo <= k:
                k -= powerOfTwo
                maxLength += 1
            if powerOfTwo <= k:
                powerOfTwo <<= 1

        return maxLength
